tools/nodeset_compiler/nodeset.py

```diff
@@ -227,16 +227,16 @@ def filterRef(r, rt):
             if r.target == node.id:
                 if r.source not in self.nodes:
                     continue
-                self.nodes[r.source].references = set(filter(
-                    lambda rt: filterRef(r, rt),
-                    self.nodes[r.source].references
+                self.nodes[r.source].references = dict(filter(
+                    lambda rt: filterRef(r, rt[0]),
+                    self.nodes[r.source].references.items()
                 ))
             elif r.source == node.id:
                 if r.target not in self.nodes:
                     continue
-                self.nodes[r.target].references = set(filter(
-                    lambda rt: filterRef(r, rt),
-                    self.nodes[r.target].references
+                self.nodes[r.target].references = dict(filter(
+                    lambda rt: filterRef(r, rt[0]),
```

Comment:
Why rt[0]?
Add a comment to document.

Reply:
only the key is passed through the filter. The dict.items statement returns all key, value pairs as views. In order to get the key, the first item must be selected.
I added a comment.